pkg/ibmcsidriver/controller_helper.go

```diff
@@ -303,8 +303,8 @@ func getVolumeParameters(logger *zap.Logger, req *csi.CreateVolumeRequest, confi
 	//If ENI/VNI enabled then check for scenarios where zone and subnetId is mandatory
 	if volume.VPCVolume.AccessControlMode == SecurityGroup {
 
-		//Zone is mandatory if subnetID or primaryIPID/primaryIPAddress is user defined
-		if len(strings.TrimSpace(volume.Az)) == 0 && (len(volume.VPCVolume.SubnetID) != 0 || (volume.VPCVolume.PrimaryIP != nil)) {
+		//Zone and Region is mandatory if subnetID or primaryIPID/primaryIPAddress is user defined
+		if (len(strings.TrimSpace(volume.Az)) == 0 || len(strings.TrimSpace(volume.Region)) == 0) && (len(volume.VPCVolume.SubnetID) != 0 || (volume.VPCVolume.PrimaryIP != nil)) {
```

Comment:
can this be optimized, in case of subnetID or primaryIPID is not given then also we are checking zone & region which is not required

Reply:
We will take in next PR and make it case wise